internal/gengapic/gengapic_test.go

```diff
@@ -499,3 +500,86 @@ func Test_isLRO(t *testing.T) {
 		}
 	}
 }
+
+func Test_transportParse(t *testing.T) {
+	for _, tst := range []struct {
+		param        string
+		expectedOpts options
+		expectErr    bool
+	}{
+		{
+			param: "transport=grpc,go-gapic-package=path;pkg",
+			expectedOpts: options{
+				transports: []Transport{grpc},
+				pkgPath:    "path",
+				pkgName:    "pkg",
+				outDir:     "path",
+			},
+			expectErr: false,
+		},
+		{
+			param: "transport=rest+grpc,go-gapic-package=path;pkg",
+			expectedOpts: options{
+				transports: []Transport{rest, grpc},
+				pkgPath:    "path",
+				pkgName:    "pkg",
+				outDir:     "path",
+			},
+			expectErr: false,
+		},
+		{
+			param: "go-gapic-package=path;pkg",
+			expectedOpts: options{
+				transports: []Transport{grpc},
+				pkgPath:    "path",
+				pkgName:    "pkg",
+				outDir:     "path",
+			},
+		},
+		{
+			param: "module=path,go-gapic-package=path/to/out;pkg",
+			expectedOpts: options{
+				transports:   []Transport{grpc},
+				pkgPath:      "path/to/out",
+				pkgName:      "pkg",
+				outDir:       "to/out",
+				modulePrefix: "path",
+			},
+			expectErr: false,
+		},
+		{
+			param:     "transport=tcp,go-gapic-package=path;pkg",
+			expectErr: true,
+		},
+		{
+			param:     "go-gapic-package=pkg;",
+			expectErr: true,
+		},
+		{
+			param:     "go-gapic-package=;path",
+			expectErr: true,
+		},
+		{
+			param:     "go-gapic-package=bogus",
+			expectErr: true,
+		},
+		{
+			param:     "module=different_path,go-gapic-package=path;pkg",
+			expectErr: true,
+		},
+	} {
+		opts, err := ParseOptions(&tst.param)
+		if tst.expectErr {
+			if err == nil {
+				t.Errorf("ParseOptions(%s) expected error", tst.param)
+			}
+		} else {
```

Comment:
idiomaticity: Go prefers exiting early and avoiding nested code. Son this case, maybe
```
if tst.ExpectErr {
 ...
  continue
}
if err != nil {
  ...
  continue  // <- you probably want this, but maybe not
}
if !reflect... {
  ...
}
}
```

Reply:
I am absolutely behind less nesting, but my first inclination is to try to do without `continue`. If my attempts in that area are unsuccessful I'll lean towards shallow+continue.

Tangent: one of my least favorite things in reading code, either code I've written or something other people have written, is complicated control flow. It's been big source of difficult to diagnose "push the bubble in the wallpaper" bugs where fixing one exposes a new problem. A big part of my python style is trying to avoid explicit control flow, and I need to retrain those muscles for golang's idioms.